sdks/java/io/google-cloud-platform/src/test/java/org/apache/beam/sdk/io/gcp/pubsub/PubsubReadSchemaTransformProviderTest.java

```diff
@@ -18,300 +18,240 @@
               .withFieldValue("name", "c")
               .withFieldValue("number", 300L)
               .build());
 
-  private static final Clock CLOCK = (Clock & Serializable) () -> 1656788475425L;
+  private static final List<Row> ROWSWITHERROR =
+      Arrays.asList(
+          Row.withSchema(BEAMSCHEMAWITHERROR).withFieldValue("error", "a").build(),
+          Row.withSchema(BEAMSCHEMAWITHERROR).withFieldValue("error", "b").build(),
+          Row.withSchema(BEAMSCHEMAWITHERROR).withFieldValue("error", "c").build());
+
+  private static final Clock CLOCK = (Clock & Serializable) () -> 1678988970000L;
 
   private static final AvroPayloadSerializerProvider AVRO_PAYLOAD_SERIALIZER_PROVIDER =
       new AvroPayloadSerializerProvider();
   private static final PayloadSerializer AVRO_PAYLOAD_SERIALIZER =
-      AVRO_PAYLOAD_SERIALIZER_PROVIDER.getSerializer(SCHEMA, new HashMap<>());
-
-  @Rule public TestPipeline p = TestPipeline.create().enableAbandonedNodeEnforcement(false);
+      AVRO_PAYLOAD_SERIALIZER_PROVIDER.getSerializer(BEAMSCHEMA, new HashMap<>());
+  private static final PayloadSerializer AVRO_PAYLOAD_SERIALIZER_WITH_ERROR =
+      AVRO_PAYLOAD_SERIALIZER_PROVIDER.getSerializer(BEAMSCHEMAWITHERROR, new HashMap<>());
 
-  @Test
-  public void testBuildDeadLetterQueueWrite() {
-    for (TestCase testCase : cases) {
-      PubsubIO.Write<PubsubMessage> dlq =
-          testCase.pubsubReadSchemaTransform().buildDeadLetterQueueWrite();
-
-      if (testCase.expectedDeadLetterQueue == null) {
-        assertNull(testCase.name, dlq);
-        return;
-      }
-
-      Map<DisplayData.Identifier, DisplayData.Item> actual = DisplayData.from(dlq).asMap();
-      Map<DisplayData.Identifier, DisplayData.Item> expected = testCase.expectedDeadLetterQueue;
-
-      assertEquals(testCase.name, expected, actual);
-    }
-  }
+  @Rule public transient TestPipeline p = TestPipeline.create();
 
   @Test
-  public void testReadAvro() throws IOException {
+  public void testInvalidConfigNoTopicOrSubscription() {
     PCollectionRowTuple begin = PCollectionRowTuple.empty(p);
-    PubsubReadSchemaTransformProvider.PubsubReadSchemaTransform transform =
-        schemaTransformWithClock("avro");
-    PubsubTestClient.PubsubTestClientFactory clientFactory =
-        clientFactory(incomingAvroMessagesOf(CLOCK.currentTimeMillis()));
-    transform.setClientFactory(clientFactory);
-    PCollectionRowTuple reads = begin.apply(transform);
-
-    PAssert.that(reads.get(PubsubReadSchemaTransformProvider.OUTPUT_TAG)).containsInAnyOrder(ROWS);
-
+    assertThrows(
+        IllegalArgumentException.class,
+        () ->
+            begin.apply(
+                new PubsubReadSchemaTransformProvider()
+                    .from(
+                        PubsubReadSchemaTransformConfiguration.builder()
+                            .setSchema(SCHEMA)
+                            .setFormat("AVRO")
+                            .build())));
     p.run().waitUntilFinish();
```

Comment:
Looks like validation checks live in PubsubReadSchemaTransformProvider::from, so we actually don't need a pipeline to trigger these exceptions (we just need a `PubsubReadSchemaTransformProvider().from(...)` call)

Reply:
Done